### ISSUE TITLE
why memory cost too much than three.js?and how to improve performance?

### ISSUE THREAD

**Issue by luojiandan** [jscastro76/threebox#55]
Hi @jscastro76 ,
Thank you for maintaining this plugin.
when I  add the same model to the map for **100 times by threebox.j**s,the memory cost **2GB**.and set the number to 500,the browser crash
but add the same model to the map for **5000 times by three.js**,the memory cost **162M**.
Why is the gap so large and how can I improve performance?

**model file format:**glb
**model size:**4M
![image](https://user-images.githubusercontent.com/5663556/96745945-3b715680-13f9-11eb-9255-a0e24db66b51.png)

**three.js:**
memory cost 162M
![image](https://user-images.githubusercontent.com/5663556/96745994-4e842680-13f9-11eb-90d2-aff9a234c9a6.png)

**threebox.js:**
```javascript
          //the options setting 
          let options = {
            type: "gltf", //'gltf'/'mtl'            
            obj: "./model/Freighter.glb",  //model file path
            bin: "", //replace by mtl attribute
            units: "meters", //units in the default values are always in meters
            scale: 30, //boat size
            rotation: { x: 90, y: 180, z: 0 }, //default rotation
            anchor: "center",
            customProperties: {
              clicked: false,
            },
          };
```
**way 1:**
![image](https://user-images.githubusercontent.com/5663556/96746194-82f7e280-13f9-11eb-8fad-a884d97f730a.png)

**way 2(duplicate):**
![image](https://user-images.githubusercontent.com/5663556/96746218-8a1ef080-13f9-11eb-8b0d-d8615fc58682.png)

Am I in the right way or there another way?
thanks again:)

**Comment by luojiandan** [comment]
[Performance considerations](https://github.com/jscastro76/threebox/blob/master/docs/Threebox.md#performance-considerations)
how to use  **obj.duplicate()** ,my code only get one model on map.what's wrong with it?
```javascript
          tb.loadObj(options, function (model) {
            mdl = model.duplicate();  //use duplicate
            origin.forEach((el) => {              
              mdl.setCoords(el).addTooltip("Boat", false);
              console.log("3 克隆", mdl);
              tb.add(mdl);
            });
          });
```

**Comment by luojiandan** [comment]
@jscastro76 
```javascript
          var origin = new Array();
          for (let index = 0; index < 100; index++) {
            var coordinates = new Array(2);
            coordinates[0] = Math.random() * (119.287 - 117.335) + 117.335;
            coordinates[1] = Math.random() * (39.413 - 38.488) + 38.488;

            origin.push(coordinates);
          }
          
          let i=1;
          tb.loadObj(options, function (model) {                     
            console.log("loop begin");
            origin.forEach((el) => {                            
              let mdl=model.duplicate().setCoords(el);
              mdl.addTooltip("Boat", false);//.addLabel("<h1>Hello World!</h1>");//  addTooltip("Boat", false);
              console.log("clone times "+i, mdl.coordinates);
              tb.add(mdl);
              i++;
            });
            console.log("loop end");
          });
        },
```

**Comment by jscastro76** [comment]
Since a few days ago I see your issues reported, thanks for your interest in the project, but let me clarify some basic rules to have a healthy and structured interaction if you pretend to receive help. 
With partial parts of code and shifting from one topic to another in every issue is hard to follow what you are trying to describe. 
Please, keep an order in each issue on and don't mix topics in a single issue, my time, as yours I guess, is limited and this is an open source project under MIT license, so don´t consider this a support service desktop. Otherwise I'll be forced to ignore and close the issues you open.

Having said so, first you mentioned you use three.js to add 5000 objects, please share the sample code you are using in comparison to the sample code of threebox, without that is impossible I can help you with it.

Second, I see you found the right way to clone objects with `duplicate()`, which was open in other issue too, good. But was that solved your memory issue?

**Comment by jscastro76** [comment]
@luojiandan check out the new v.2.0.6 as it includes an important change in `loadObj` method. Now is not anymore needed to clone explicitly 3D loaded models as `loadObj` method manages it automatically through the internal cache.
I have added a new example that shows how to create 1000 objects in miliseconds.
